AdminNotice.php

```diff
@@ -484,10 +506,28 @@ public function isDismissed() {
 			}
 
 			if ($this->dismissionScope === self::DISMISS_PER_SITE) {
-				return (boolean)(get_option($this->getDismissOptionName(), false));
+				$dismissal = get_option($this->getDismissOptionName());
 			} else {
-				return (boolean)(get_user_meta(get_current_user_id(), $this->getDismissOptionName(), true));
+				$dismissal = get_user_meta(get_current_user_id(), $this->getDismissOptionName(), true);
 			}
+
+			$dismissal = json_decode($dismissal, true);
+
+			if (!$dismissal) {
+				return false;
+			}
+
+			// If the library has been updated from version 1.0, then the dismiss option
+			// will contain '1' instead of the dismissal time. The best we can do
+			// in this case is set the dismissal time to the current time so that
+			// the notice can at least eventually be undismissed.
+			if ($dismissal === 1) {
```

Comment:
Since `$dismissal` is no longer cast to `int`, it would probably be a string here and not a number.
```suggestion
			if ($dismissal === '1') {
```

Reply:
`json_decode()` converts it into an integer.